corehq/apps/app_manager/suite_xml/post_process/remote_requests.py

```diff
@@ -164,11 +166,11 @@ def _get_multi_select_exclude(self):
     def get_post_relevant(self):
         case_not_claimed = self.module.search_config.get_relevant(
             self.case_session_var, self.module.is_multi_select())
-        if module_uses_smart_links(self.module):
-            case_in_project = self._get_smart_link_rewind_xpath()
-            return XPath.and_(case_not_claimed, case_in_project)
-        else:
-            return case_not_claimed
+        case_in_project = self._get_smart_link_rewind_xpath()
+        uses_smart_links = module_uses_smart_links(self.module)
+        if case_search_sync_cases_on_form_entry_enabled_for_domain(self.domain):
```

Comment:
@snopoke Do we still want to remove relevant for `remote_request.post` now that we are adding `post` in `entry` blocks ? I feel we can avoid the 2 `post` requests with atleast the local cases with this workflow.

Reply:
@shubham1g5 that makes sense. 

@stephherbers FYI

Reply:
We still want to remove the condition for for the post in the entry (if `case_search_sync_cases_on_form_entry_enabled_for_domain` is true).

I think we could add a parameter to `RemoteRequestFactory` to indicate whether to include it or not.

Reply:
@snopoke is this more what you were thinking? https://github.com/dimagi/commcare-hq/pull/32554#:~:text=for%20remote_request.post-,e77d054,-Merge%20state